lib/ClangImporter/DWARFImporter.cpp

```diff
@@ -0,0 +1,178 @@
+//===--- DWARFImporter.cpp - Import Clang modules from DWARF --------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#include "ImporterImpl.h"
+#include "swift/AST/Module.h"
+
+using namespace swift;
+
+/// Represents a Clang module that was "imported" from debug info. Since all the
+/// loading of types is done on demand, this class is effectively empty.
+class DWARFModuleUnit final : public LoadedFile {
+  ~DWARFModuleUnit() = default;
+  ClangImporter::Implementation &owner;
+
+public:
+  DWARFModuleUnit(ModuleDecl &M, ClangImporter::Implementation &owner)
+      : LoadedFile(FileUnitKind::DWARFModule, M), owner(owner) {}
+
+  virtual bool isSystemModule() const override { return false; }
+
+  /// Forwards the request to the ClangImporter, which forwards it to the
+  /// DWARFimporterDelegate.
+  virtual void
+  lookupValue(ModuleDecl::AccessPathTy accessPath, DeclName name,
+              NLKind lookupKind,
+              SmallVectorImpl<ValueDecl *> &results) const override {
+    owner.lookupValueDWARF(accessPath, name, lookupKind, results);
```

Comment:
Hm, this isn't going to filter by module. Is that okay? (Is that actually desirable and I just don't have the whole model paged in?)

Reply:
https://github.com/apple/swift/pull/26707